jmxtrans-core/src/main/java/com/googlecode/jmxtrans/connections/JMXConnection.java

```diff
@@ -32,20 +34,49 @@
 import javax.management.remote.JMXConnector;
 import java.io.Closeable;
 import java.io.IOException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 @ToString
 @ThreadSafe
 public class JMXConnection implements Closeable {
 	@Nullable private final JMXConnector connector;
 	@Nonnull @Getter private final MBeanServerConnection mBeanServerConnection;
+	@Nonnull private final ExecutorService executor;
+	private boolean markedAsDestroyed;
+	private static final Logger logger = LoggerFactory.getLogger(JMXConnection.class);
 
 	public JMXConnection(@Nullable JMXConnector connector, @Nonnull MBeanServerConnection mBeanServerConnection) {
 		this.connector = connector;
 		this.mBeanServerConnection = mBeanServerConnection;
+		this.markedAsDestroyed = false;
+
+		executor = Executors.newSingleThreadExecutor();
```

Comment:
We should probably extract that executor at least as high as `MBeanServerConnectionFactory` and share it between connections. Keeping a single thread executor is probably fine for this case (if we have enough servers in trouble to saturate the single thread executor, we probably want to apply some back pressure to the connection pool anyway, we can tune that in the future if need be).

This would also allow the use of a sameThreadExecutor in the test, slightly simpler.

Reply:
Yeah, I thought about it. I imagined this situation. It will calls rarely.
But your sentece sounds good. I will try to fix it.